source/aws_profile.c

```diff
@@ -862,6 +863,31 @@ static struct aws_byte_cursor s_trim_trailing_whitespace_comment(const struct aw
     return trimmed;
 }
 
+static int s_parse_section_type_prefix(
+    struct aws_byte_cursor *profile_cursor,
+    enum aws_profile_section_type *out_section_type) {
+    /* check profile prefix */
+    if (s_parse_by_token(profile_cursor, s_profile_token, NULL) &&
+        s_parse_by_character_predicate(profile_cursor, s_is_whitespace, NULL, 1)) {
+        *out_section_type = AWS_PROFILE_SECTION_TYPE_PROFILE;
+        return AWS_OP_SUCCESS;
+    }
+    /* check sso-session prefix */
+    if (s_parse_by_token(profile_cursor, s_sso_session_token, NULL) &&
+        s_parse_by_character_predicate(profile_cursor, s_is_whitespace, NULL, 1)) {
```

Comment:
s_is_whitespace() also allows '\n' which is .... maybe not what we want here
```
[profile
bozo]
```
shouldn't pass, right?

Reply:
Thanks, this is not a problem since we split the config file by \n first https://github.com/awslabs/aws-c-sdkutils/blob/db8625aea8c38861d1005fe7869cc63909ba454b/source/aws_profile.c#L1285. So
```
[profile
bozo]
```
is invalid since it will be treated as 2 individual lines.